packages/graphql/src/schema-model/annotation/AuthorizationAnnotation.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const AuthorizationAnnotationArguments = {
+    filter: "filter",
+    validate: "validate",
+};
+
+export type AuthorizationFilterOperation = "READ" | "UPDATE" | "DELETE" | "CREATE_RELATIONSHIP" | "DELETE_RELATIONSHIP";
+
+export type AuthorizationValidateOperation =
+    | "READ"
+    | "CREATE"
+    | "UPDATE"
+    | "DELETE"
+    | "CREATE_RELATIONSHIP"
+    | "DELETE_RELATIONSHIP";
+
+export type ValidateWhen = "BEFORE" | "AFTER";
+
+export const AuthorizationFilterOperationRule: ReadonlyArray<AuthorizationFilterOperation> = [
+    "READ",
+    "UPDATE",
+    "DELETE",
+    "CREATE_RELATIONSHIP",
+    "DELETE_RELATIONSHIP",
+];
+
+export const AuthorizationValidateOperationRule: ReadonlyArray<AuthorizationValidateOperation> = [
+    "READ",
+    "CREATE",
+    "UPDATE",
+    "DELETE",
+    "CREATE_RELATIONSHIP",
+    "DELETE_RELATIONSHIP",
+];
```

Comment:
Why did we decide against inferring the types from these declared as a `const` array in the end? Seems to be that we now have duplicate specification of these strings now.

Reply:
It could be inferred directly but as in another comment was pointed out, it was less readable, after the change I agree that even if a bit redundant is way easier to understand this type.